src/tests/krate.rs

```diff
@@ -1589,6 +1589,120 @@ fn publish_after_yank_max_version() {
     assert_eq!(json.krate.max_version, "2.0.0");
 }
 
+#[test]
+fn publish_records_an_audit_action() {
+    use cargo_registry::models::{VersionAction, VersionOwnerAction};
+
+    let (app, anon, _, token) = TestApp::full().with_token();
+
+    app.db(|conn| assert!(VersionOwnerAction::all(&conn).unwrap().is_empty()));
+
+    // Upload a new crate, putting it in the git index
+    let crate_to_publish = PublishBuilder::new("fyk");
+    token.enqueue_publish(crate_to_publish).good();
+
+    // make sure it doesn't have any audit actions
+    // do this as a full integration test once the api is in place.
+    let json = anon.show_version("fyk", "1.0.0");
+    let version_id = json.version.id;
+
+    // make sure it has one audit action
+    // do this as a full integration test once the api is in place.
+    app.db(|conn| {
+        let actions = VersionOwnerAction::all(conn).unwrap();
+        assert_eq!(actions.len(), 1);
+        let action = actions[0];
+        assert_eq!(action.version_id, version_id);
+        assert_eq!(action.user_id, token.as_model().user_id);
+        assert_eq!(action.api_token_id, Some(token.as_model().id));
+        assert_eq!(action.action, VersionAction::Publish);
+    });
+}
+
+#[test]
+fn yank_records_an_audit_action() {
+    use cargo_registry::models::{VersionAction, VersionOwnerAction};
+
+    let (app, anon, _, token) = TestApp::full().with_token();
+
+    // Upload a new crate, putting it in the git index
+    let crate_to_publish = PublishBuilder::new("fyk");
+    token.enqueue_publish(crate_to_publish).good();
+
+    // make sure it doesn't have any audit actions
```

Comment:
Same here, this is confusing because this version will have 1 audit action (publish) at this point...